pkg/pillar/types/zedagenttypes.go

```diff
@@ -382,19 +382,42 @@ func BootReasonFromString(str string) BootReason {
 	}
 }
 
+// MaintenanceModeReason captures reason for entering into maintenance mode
+type MaintenanceModeReason uint8
+
+// MaintenanceModeReason codes for storing reason for getting into maintenance mode
+const (
+	MaintenanceModeReasonNone MaintenanceModeReason = iota
+	MaintenanceModeReasonVaultLockedUp
+)
+
+// String returns the verbose equivalent of MaintenanceModeReason code
+func (mmr MaintenanceModeReason) String() string {
+	switch mmr {
+	case MaintenanceModeReasonNone:
+		return "MaintenanceModeReasonNone"
+	case MaintenanceModeReasonVaultLockedUp:
+		return "MaintenanceModeReasonVaultLockedUp"
+	default:
+		return "MaintenanceModeReasonUnknown"
```

Comment:
In similar code I've done a fmt.Sprintf to print the unknow value with %d.

Reply:
@eriknordmark addressed your comments, please take a look now.